tests/tape/test_tape.py

```diff
@@ -860,7 +860,7 @@ def test_array_parameter(self):
         params = [a, 0.32, 0.76, 1.0]
 
         with QuantumTape() as tape:
-            op = qml.QubitStateVector(params[0], wires=0)
+            op = qml.QubitStateVector(params[0], wires=[0, 1])
```

Comment:
Was it changed because an error was raised or because you wanted to test the reordering it inside a tape?

Reply:
error was raised. if wires has length $n$, the provided state must have length $2^n$

Reply:
Before your could apply it on a subset of wires and assume everything else is zero, right?

Reply:
nope, the logic used to be in`DefaultQubit._apply_state_vector()` (which was called in `DefaultQubit.apply()`), meaning the error would only be raised when it is being applied by a device. Now, I raise the error at construction-time in `QubitStateVector.__init__()` because it was not a legitimate Operator at that point.

As for the everything-else-is-zero assumption, applying `QubitStateVector` _used to_ fail if it didn't touch all wires on the device it's being applied to. With this PR, it makes the zero-assumption for missing wires as an added functionality. I started implementing that before realizing that it was an error on the old device, but now that I type it out, I wonder if maybe we don't want to add that feature?